src/interface/src/app/login/login.component.spec.ts

```diff
@@ -0,0 +1,54 @@
+import { NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { FormsModule } from '@angular/forms';
+import { Router } from '@angular/router';
+import { of } from 'rxjs';
+
+import { AuthService } from './../auth.service';
+import { LoginComponent } from './login.component';
+
+describe('LoginComponent', () => {
+  let component: LoginComponent;
+  let fixture: ComponentFixture<LoginComponent>;
+
+  beforeEach(() => {
+    const routerStub = () => ({ navigate: (array: string[]) => ({}) });
+    const fakeAuthService = jasmine.createSpyObj<AuthService>(
+      'AuthService',
+      { login: of({}) },
+      {},
+    );
+    TestBed.configureTestingModule({
+      imports: [FormsModule],
+      schemas: [NO_ERRORS_SCHEMA],
+      declarations: [LoginComponent],
+      providers: [
+        { provide: Router, useFactory: routerStub },
+        { provide: AuthService, useValue: fakeAuthService }
+      ]
+    });
+    fixture = TestBed.createComponent(LoginComponent);
+    component = fixture.componentInstance;
+  });
+
+  it('can load instance', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('onSubmit', () => {
+    it('makes expected calls', () => {
+      spyOn(component, 'login').and.callThrough();
+      component.onSubmit();
+      expect(component.login).toHaveBeenCalled();
+    });
+  });
+
+  describe('signup', () => {
+    it('makes expected calls', () => {
```

Comment:
nit: more specific it-block description? "navigates to signup"? same with other "makes expected calls" tests.

Reply:
Done!